### ISSUE TITLE
castOnHydrate strange behaviour

### ISSUE THREAD

**Issue by Shay12tg** [phalcon/cphalcon#12199]
Hi.

I tried using castOnHydrate but I can't get it to work properly.

In PHP 7.0.8 I get: 
Column 'id' is not part of the column map
I tried to disable and change model caching methods but no change.

In PHP 5.6.24 It's working...ish, it only cast if you select one result ( findFirst ) but it doesn't if you use find()->toArray

I also found this unanswered post: https://forum.phalconphp.com/discussion/3389/column-storage-is-not-part-of-column-map
And this issue: https://github.com/phalcon/cphalcon/issues/11205
### Details

Phalcon version: 3.0.1
PHP Version: 5.6 and 7.0
Operating System: Ubuntu 16.04
Tried both compiling from source and installing via package manager
Zephir version: 0.9.4a-dev-7e304ba18c
Server:  Apache 2.4
DB: MySQL 5.6

**Comment by pfz** [comment]
It should have been fixed in 3.0.2, see #12147 .

**Comment by sergeyklay** [comment]
@Shay12tg 
Could you please check 3.0.x branch?

```
git clone git@github.com:phalcon/cphalcon.git
cd cphalcon
git checkout 3.0.x

zephir fullclean
zephir build
```
